src/mongoc/mongoc-write-command.c

```diff
@@ -698,13 +718,22 @@ _mongoc_write_command_update (mongoc_write_command_t       *command,
     * opcodes, then submit the legacy opcode so we don't need to wait for
     * a response from the server.
     */
-   if ((client->cluster.nodes [hint - 1].min_wire_version == 0) &&
+
+   server = mongoc_topology_server_by_id(client->topology, hint);
+
+   if (!server) {
+      EXIT;
+   }
+
+   if ((server->min_wire_version == 0) &&
        !_mongoc_write_concern_needs_gle (write_concern)) {
       _mongoc_write_command_update_legacy (command, client, hint, database,
                                            collection, write_concern, result,
                                            error);
+      mongoc_server_description_destroy (server);
```

Comment:
maybe move this to a cleanup label that would go on line 760 and then GOTO(cleanup) here? It would avoid the second call to mongoc_server_description_destroy, which is a bit hacky IMO.